loadbalancer/zk/client.go

```diff
@@ -0,0 +1,157 @@
+package zk
+
+import (
+	"errors"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/eapache/channels"
+	"github.com/samuel/go-zookeeper/zk"
+
+	"github.com/go-kit/kit/log"
+)
+
+// DefaultACL is the default ACL to use for creating znodes
+var DefaultACL = zk.WorldACL(zk.PermAll)
+
+const (
+	// ConnectTimeout is the default timeout to establish a connection to a
+	// ZooKeeper node.
+	ConnectTimeout = 2 * time.Second
+	// SessionTimeout is the default timeout to keep the current ZooKeeper
+	// session alive during a temporary disconnect.
+	SessionTimeout = 5 * time.Second
+)
+
+// Client is a wrapper around a lower level ZooKeeper client implementation.
+type Client interface {
+	// GetEntries should query the provided path in ZooKeeper, place a watch on
+	// it and retrieve data from its current child nodes.
+	GetEntries(path string) ([]string, channels.SimpleOutChannel, error)
```

Comment:
To allow for a generic Client interface so people can swap out the existing client implementation in favor of another and as long as it will satisfy the Client interface no code change is needed for the ZooKeeper publisher. The channels package has decent wrap and unwrap functions and other channel helper functions so I used that instead of rolling my own. If we don't want the ability to swap implementations the entire Client interface could be left out but we end up with a tightly coupled Publisher.